app/src/main/java/com/duckduckgo/app/onboarding/ui/page/WelcomePageViewModel.kt

```diff
@@ -33,16 +42,52 @@ class WelcomePageViewModel(
     private val context: Context,
     private val pixel: Pixel,
     private val defaultRoleBrowserDialog: DefaultRoleBrowserDialog,
+    private val variantManager: VariantManager,
 ) : ViewModel() {
 
     fun reduce(event: WelcomePageView.Event): Flow<WelcomePageView.State> {
         return when (event) {
+            WelcomePageView.Event.ShowFirstDaxOnboardingDialog -> showFirstDaxOnboardingDialog()
+            WelcomePageView.Event.OnSkipOptions -> onSkipOptionsClicked()
+            is WelcomePageView.Event.OnContinueOptions -> onContinueWithOptionsClicked(event.options)
             WelcomePageView.Event.OnPrimaryCtaClicked -> onPrimaryCtaClicked()
             WelcomePageView.Event.OnDefaultBrowserSet -> onDefaultBrowserSet()
             WelcomePageView.Event.OnDefaultBrowserNotSet -> onDefaultBrowserNotSet()
         }
     }
 
+    private fun showFirstDaxOnboardingDialog() = flow {
+        when (variantManager.isOnboardingCustomizationExperimentEnabled()) {
+            true -> emit(WelcomePageView.State.ShowFeatureOptionsCta)
+            false -> emit(WelcomePageView.State.ShowControlDaxCta)
+        }
+    }
+
+    private fun onSkipOptionsClicked(): Flow<WelcomePageView.State> = flow {
+        pixel.fire(AppPixelName.ONBOARDING_OPTION_SKIP)
+        emit(WelcomePageView.State.ShowControlDaxCta)
+    }
+
+    private fun onContinueWithOptionsClicked(options: Map<DDGFeatureOnboardingOption, Boolean>): Flow<WelcomePageView.State> = flow {
+        val optionsSelected = options.filter { it.value }.map { it.key }.toList()
+        if (optionsSelected.isEmpty()) {
+            pixel.fire(AppPixelName.ONBOARDING_OPTION_SKIP)
```

Comment:
I thought this was not possible, but testing I see it's possible to select and unselect again any option, leaving the primary button enabled without anything selected. I assume we want that, but sharing I would have expected the primary to be disabled again.